pkgs/development/python-modules/tree-sitter-grammars/default.nix

```diff
@@ -0,0 +1,128 @@
+{ lib
+, buildPythonPackage
+, pytestCheckHook
+, tree-sitter
+, symlinkJoin
+, writeTextDir
+  # `name`: grammar derivation pname in the format of `tree-sitter-<lang>`
+, name
+, grammarDrv
+}:
+let
+  inherit (grammarDrv) version;
+
+  snakeCaseName = lib.replaceStrings [ "-" ] [ "_" ] name;
+  drvPrefix = "python-${name}";
+  langIdentOverrides = {
+    tree_sitter_org_nvim = "tree_sitter_org";
+  };
+  langIdent = langIdentOverrides.${snakeCaseName} or snakeCaseName;
+in
+assert (lib.assertMsg (!(grammarDrv.meta ? license)) ''
+  As of this writing, ${grammarDrv.pname} surprisingly doesn't have a license.
+  This trap is set here to guarantee that if it ever does have a license, this package will inherit the license.
+'');
+buildPythonPackage {
+  inherit version;
+  pname = drvPrefix;
+
+  src = symlinkJoin {
```

Comment:
It would probably be cleaner keeping the files in-tree, pointing `src` there and using `substituteInPlace --subst-var-by` in `postPatch` to pass the variables.

Reply:
Hi, that is indeed a solution. However, note that two files are generated from the variable: ${snakeCaseName}/__init__.py and ${snakeCaseName}/binding.c.

This may require some file movement. Perhaps a hybrid solution could work as well.